src/core/addonManager.js

```diff
@@ -123,15 +125,21 @@ export function install(
   return _mozAddonManager.createInstall({ url, hash }).then((installObj) => {
     const callback = (e) => eventCallback(installObj, e);
     for (const event of INSTALL_EVENT_LIST) {
-      log.info(`[install] Adding listener for ${event}`);
+      _log.info(`[install] Adding listener for ${event}`);
       installObj.addEventListener(event, callback);
     }
     return new Promise((resolve, reject) => {
       installObj.addEventListener('onInstallEnded', () => resolve());
       installObj.addEventListener('onInstallFailed', () => reject());
-      log.info('Events to handle the installation initialized.');
-      // See: https://github.com/mozilla/addons-frontend/issues/8633
-      installObj.install().catch(reject);
+      _log.info('Events to handle the installation initialized.');
+
+      installObj.install().catch((error) => {
+        // The `mozAddonManager` has events we can listen to, this error occurs
+        // when a user cancels the installation but we are already notified via
+        // `onInstallCancelled`.
+        // See: https://github.com/mozilla/addons-frontend/issues/8668
+        _log.warn(`Ignoring promise rejection during installation: ${error}`);
```

Comment:
It's a little confusing that `log.warn()` has a special meaning, only for the tests. I think I'd rather see it done explicitly. What are your thoughts on using an explicit callback?

```diff
diff --git a/src/core/addonManager.js b/src/core/addonManager.js
index 0247f0089..5998fcbf6 100644
--- a/src/core/addonManager.js
+++ b/src/core/addonManager.js
@@ -105,6 +105,7 @@ type OptionalInstallParams = {|
   _log?: typeof log,
   src: string,
   hash?: string | null,
+  onIgnoredRejection?: () => void,
 |};
 
 export function install(
@@ -114,6 +115,7 @@ export function install(
     _log = log,
     _mozAddonManager = window.navigator.mozAddonManager,
     hash,
+    onIgnoredRejection = () => {},
     src,
   }: OptionalInstallParams = {},
 ) {
@@ -135,7 +137,7 @@ export function install(
       installObj.install().catch((error) => {
         // See: https://github.com/mozilla/addons-frontend/issues/8668
         _log.warn(`Ignoring error during installation: ${error}`);
-        resolve();
+        onIgnoredRejection();
       });
     });
   });
diff --git a/tests/unit/core/test_addonManager.js b/tests/unit/core/test_addonManager.js
index f7436c4c4..89fa2a566 100644
--- a/tests/unit/core/test_addonManager.js
+++ b/tests/unit/core/test_addonManager.js
@@ -199,17 +199,27 @@ describe(__filename, () => {
 
     // See: https://github.com/mozilla/addons-frontend/issues/8633
     it('logs and ignores rejected install errors', async () => {
+      let finishInstall;
+      const installToFinish = new Promise((resolve) => {
+        finishInstall = resolve;
+      });
+
       const _log = getFakeLogger();
+
       fakeInstallObj.install = sinon.spy(() => {
         return Promise.reject(new Error('oops'));
       });
 
-      await addonManager.install(fakeInstallUrl, fakeCallback, {
-        _log,
-        _mozAddonManager: fakeMozAddonManager,
-        src: 'home',
-      });
+      addonManager
+        .install(fakeInstallUrl, fakeCallback, {
+          _log,
+          _mozAddonManager: fakeMozAddonManager,
+          onIgnoredRejection: () => finishInstall(),
+          src: 'home',
+        })
+        .then(unexpectedSuccess);
 
+      await installToFinish;
       sinon.assert.calledOnce(_log.warn);
     });
 
```

Reply:
I dont like having to pass a prop only for the test case but that's more explicit, so I like it better.